crates/katana/rpc/src/api/starknet.rs

```diff
@@ -16,73 +16,121 @@ use katana_rpc_types::transaction::{
-    InvalidMessageSelector = 21,
+    InvalidMessageSelector,
     #[error("Invalid call data")]
-    InvalidCallData = 22,
+    InvalidCallData,
     #[error("Block not found")]
-    BlockNotFound = 24,
+    BlockNotFound,
     #[error("Transaction hash not found")]
-    TxnHashNotFound = 29,
+    TxnHashNotFound,
     #[error("Invalid transaction index in a block")]
-    InvalidTxnIndex = 27,
+    InvalidTxnIndex,
     #[error("Class hash not found")]
-    ClassHashNotFound = 28,
+    ClassHashNotFound,
     #[error("Requested page size is too big")]
-    PageSizeTooBig = 31,
+    PageSizeTooBig,
     #[error("There are no blocks")]
-    NoBlocks = 32,
+    NoBlocks,
     #[error("The supplied continuation token is invalid or unknown")]
-    InvalidContinuationToken = 33,
+    InvalidContinuationToken,
     #[error("Contract error")]
-    ContractError = 40,
+    ContractError { revert_error: String },
     #[error("Invalid contract class")]
-    InvalidContractClass = 50,
+    InvalidContractClass,
     #[error("Class already declared")]
-    ClassAlreadyDeclared = 51,
+    ClassAlreadyDeclared,
     #[error("Invalid transaction nonce")]
-    InvalidTransactionNonce = 52,
+    InvalidTransactionNonce,
     #[error("Max fee is smaller than the minimal transaction cost (validation plus fee transfer)")]
-    InsufficientMaxFee = 53,
+    InsufficientMaxFee,
     #[error("Account balance is smaller than the transaction's max_fee")]
-    InsufficientAccountBalance = 54,
+    InsufficientAccountBalance,
     #[error("Account validation failed")]
-    ValidationFailure = 55,
+    ValidationFailure,
     #[error("Compilation failed")]
-    CompilationFailed = 56,
+    CompilationFailed,
     #[error("Contract class size is too large")]
-    ContractClassSizeIsTooLarge = 57,
+    ContractClassSizeIsTooLarge,
     #[error("Sender address in not an account contract")]
-    NonAccount = 58,
+    NonAccount,
     #[error("A transaction with the same hash already exists in the mempool")]
-    DuplicateTransaction = 59,
+    DuplicateTransaction,
     #[error("The compiled class hash did not match the one supplied in the transaction")]
-    CompiledClassHashMismatch = 60,
+    CompiledClassHashMismatch,
     #[error("The transaction version is not supported")]
-    UnsupportedTransactionVersion = 61,
+    UnsupportedTransactionVersion,
     #[error("The contract class version is not supported")]
-    UnsupportedContractClassVersion = 62,
+    UnsupportedContractClassVersion,
     #[error("An unexpected error occured")]
-    UnexpectedError = 63,
+    UnexpectedError,
     #[error("Too many storage keys requested")]
-    ProofLimitExceeded = 10000,
+    ProofLimitExceeded,
     #[error("Too many keys provided in a filter")]
-    TooManyKeysInFilter = 34,
+    TooManyKeysInFilter,
     #[error("Failed to fetch pending transactions")]
-    FailedToFetchPendingTransactions = 38,
+    FailedToFetchPendingTransactions,
+}
+
+impl From<StarknetApiError> for i32 {
```

Comment:
we can have this as a method of `StarknetApiError` so that we dont have to clone the `revert_error` dont below

```rust
impl StarknetApiError {
    fn code(&self) -> i32 { .. }
}
```